core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -922,6 +922,15 @@ private[spark] object SparkSubmitUtils {
 
         // A Module descriptor must be specified. Entries are dummy strings
         val md = getModuleDescriptor
+        // clear ivy resolution from previous launches. The resolution file is usually at
+        // ~/.ivy2/org.apache.spark-spark-submit-parent-default.xml. In between runs, this file
```

Comment:
Hi Josh, I believe the filename comes from the combination of these lines:
https://github.com/apache/spark/blob/d1fea41363c175a67b97cb7b3fe89f9043708739/core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala#L979
and 
https://github.com/apache/spark/blob/d1fea41363c175a67b97cb7b3fe89f9043708739/core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala#L1005

Therefore it should always be the same.